src/main/drivers/at32/serial_uart_impl_mcu.h

```diff
@@ -0,0 +1,47 @@
+/*
+ * This file is part of Betaflight.
+ *
+ * Betaflight is free software. You can redistribute this software
+ * and/or modify this software under the terms of the GNU General
+ * Public License as published by the Free Software Foundation,
+ * either version 3 of the License, or (at your option) any later
+ * version.
+ *
+ * Betaflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+// Configuration constants
+#define UARTDEV_COUNT_MAX 8  // UARTs 1 to 5 + LPUART1 (index 9)
+#define UARTHARDWARE_MAX_PINS 5
+#ifndef UART_RX_BUFFER_SIZE
+#define UART_RX_BUFFER_SIZE     256
+#endif
+#ifndef UART_TX_BUFFER_SIZE
+#ifdef USE_MSP_DISPLAYPORT
+#define UART_TX_BUFFER_SIZE     1280
+#else
+#define UART_TX_BUFFER_SIZE     256
+#endif
+#endif
+
+// For at32f435/7 DmaChannel is the dmamux ,need to call dmamuxenable using dmamuxid
+#define UART_MUX_CAPABLE
+#define UART_AF_CAPABLE
```

Comment:
This define means that AF is needed on per-pin basis. IIRC F4 uses single AF per port (or UARTs)

_CAPABLE sounds like something optional (perfect for pin swap).

Maybe UART_PIN_AF_NEEDED, UART_DMAMUX_NEEDED ? Or UART_USES_PIN_AF, UART_USES_DMAMUX